lib/matplotlib/hatch.py

```diff
@@ -182,6 +182,8 @@ def __init__(self, hatch, density):
 def _validate_hatch_pattern(hatch):
     valid_hatch_patterns = set(r'-+|/\xXoO.*')
     if hatch is not None:
+        if not isinstance(hatch, str):
```

Comment:
This is an subtle API change that needs to be document as currently 

```python
b.set_hatch(('x','x'))
```

works.  The docstring on `Patch.set_hatch` is not clear about the type on the input should be.

We do run the hatches through an lrucache so only hashable things work (and it looks like we used as a key in a dictionary before that).

Reply:
Working on it, 
I'm a bit lost on -
>We do run the hatches through an **lrucache** so only hashable things work (and it looks like we used as a key in a dictionary before that).

Could you break it down for me

Reply:
https://github.com/matplotlib/matplotlib/blob/199c31fa64f46d2b3dff59b0921eb0472f72570b/lib/matplotlib/path.py#L1025-L1035

Is part of the code path between `Patch.set_hatch` and rendering a hatched patch.  The `lru_cache` means we will avoid re-computing the hatch path.  The `lru_cache` is backed by a dictionary (eventually) with the function parameters as the key so only inputs that are hashable can be used.